### PR TITLE
i#4014 dr$sim phys: Export open-address hashtable from DR

### PR DESCRIPTION
**Pull request by derekbruening**
For virtual-to-physical translation, we cannot use a 3rd-party library
such as STL due to static linking constraints.  Yet the drcontainers
hashtable performs poorly; we need an open-address hashtable.  Since
DR has one we export it here in a new interface.

Adds a simple test and documentation.

Issue: #4014

### CODE REVIEW

**Comment by derekbruening** [comment]
tool.drcachesim.scattergather is the 32-bit failure: #5329

File: ext/drcontainers/hashtable.h [hunk 1]
```diff
@@ -153,6 +153,8 @@ hashtable_init(hashtable_t *table, uint num_bits, hash_type_t hashtype, bool str
  * @param[in]  cmp_key_func        A callback for comparing two keys.
  *   Leave it NULL if no callback is needed and the default is to be used.
  *   For HASH_CUSTOM, a callback must be provided.
+ *
```
Comment (abhinav92003) [inline ext/drcontainers/hashtable.h:156]: Mention explicitly above that this one uses closed addressing.

Comment (derekbruening) [reply ext/drcontainers/hashtable.h:156]: Done.

File: api/docs/release.dox [hunk 2]
```diff
@@ -140,6 +140,10 @@ Further non-compatibility-affecting changes include:
    choices under the -raw_compress option.  Compressing with lz4 is now the
    default (if built with lz4 support).
  - Added drmodtrack_lookup_pc_from_index().
+ - Added an open-address hashtable implementation for cases where third-party
+   library must be avoided and open addressing is best: dr_hashtable_create(),
```
[the remark this answers is not in the document]
Comment (derekbruening) [reply api/docs/release.dox:144]: Profiling showed all the time spent on the extra load in the drcontainers hashtable.  Yes, the intuition is that the v2p has millions of lookups and no removals.

[the remark this answers is not in the document]
Comment (derekbruening) [reply api/docs/release.dox:144]: Done: s/library/libraries/

**Comment by derekbruening** [comment]
These flakes are getting more common: #5514 phys-threads, #4604 32-bit api.startstop sysenter.

**Pull request merged**